app/helpers/pieces_helper.rb

```diff
@@ -0,0 +1,14 @@
+module PiecesHelper
+  def castling(destination_piece)
+    unless destination_piece.nil?
+      if current_piece.type == 'King' && destination_piece.type == 'Rook'
```

Comment:
This line/logic feels like it belongs in the model method.

What I would do. Is in `Piece`.

Define this:
```
#piece.rb
def can_castle?
  false
end
```

Then for all pieces, the answer is false.

Except in King, where you have defined it. (Which will override the definition in Piece).

Then in king, I'd update it to look like this.

```
def can_castle?(piece)
  return false unless piece.is_a? Rook
  return false unless first_move?
  #...ect...
end
```

Then you can delete this line
```
if current_piece.type == 'King' && destination_piece.type == 'Rook'
```